client/pages/sections/protocols/steps.js

```diff
@@ -247,7 +247,7 @@ class Step extends Component {
       >
         <NewComments comments={relevantComments} />
         {
-          !values.deleted && <StepBadge fields={values} changeFieldPrefix={changeFieldPrefix} protocolId={protocol.id} position={index}/>
+          !values.deleted && !pdf && <StepBadge fields={values} changeFieldPrefix={changeFieldPrefix} protocolId={protocol.id} position={index}/>
```

Comment:
The ChangeBadge code handles this by checking if the arrays are valid https://github.com/UKHomeOffice/asl-projects/blob/master/client/components/changed-badge.js#L23-L32. I don't think we can rely on all of them being set if we're not rendering a pdf - so I think it would be better to check in the StepBadge and return null if the previous step arrays aren't defined.